1-raw-data/mailinglists/googlegroups/googlegroups_data.go

```diff
@@ -0,0 +1,429 @@
+					dateToParse = matchDate
+				}
+				if fileName, err = getFileName(dateToParse); err != nil {
+					err = fmt.Errorf("Defining month year string key for topic lists returned an error: %v", err)
+					return
+				}
+
+				msgURL = fmt.Sprintf("https://groups.google.com/forum/?_escaped_fragment_=topic/%s/%s", groupName, topicID)
+
+				if dom, err = httpDomResponse(msgURL); err != nil {
+					return
+				}
+
+				// Get the message ids from the links associated with each topic id
+				if rawMsgURL, err = getMsgIDsFromDom(org, topicID, groupName, dom); err != nil {
+					return
+				}
+				// Store the urls for the raw message content into a map grouped by year-month
+				rawMsgUrlMap[fileName] = append(rawMsgUrlMap[fileName], rawMsgURL)
+			}
+		})
+	})
+	return
+}
+
+// Worker converting list of topic ids urls to dom objects and converting topic ids into the raw message urls
+func getRawMsgURLWorker(org, groupName string, topicURLJobs <-chan string, results chan<- urlResults) {
+	var (
+		dom                      *goquery.Document
+		topicResults, tmpResults map[string][]string
+		err                      error
+	)
+	topicResults = make(map[string][]string)
+	tmpResults = make(map[string][]string)
+
+	for url := range topicURLJobs {
+		if dom, err = httpDomResponse(url); err != nil {
+			results <- urlResults{err: err}
+			return
+		}
+
+		if tmpResults, err = topicIDToRawMsgUrlMap(org, groupName, dom); err != nil {
+			results <- urlResults{err: fmt.Errorf("Getting dom info returned an error: %v", err)}
+			return
+		}
+
+		for fileName, rawMsgURL := range tmpResults {
+			topicResults[fileName] = append(topicResults[fileName], rawMsgURL...)
+			log.Printf("Filename results grabbed %s.", fileName)
+		}
+
+	}
+	results <- urlResults{urlMap: topicResults, err: nil}
+	return
+}
+
+// Goroutine setup to get/consolidate list of raw message urls by year-month text filename for pages with lists of topic urls.
+func listRawMsgURLsByMonth(org, groupName string, worker int) (rawMsgUrlMap map[string][]string, err error) {
+	var (
+		urlTopicList                        string
+		pageIndex, countMsgs, totalMessages int
+		dom                                 *goquery.Document
+	)
+
+	pageIndex, countMsgs = 0, 0
+	rawMsgUrlMap = make(map[string][]string)
+
+	urlTopicList = fmt.Sprintf("https://groups.google.com%s/forum/?_escaped_fragment_=forum/%s", org, groupName)
+
+	//Get total topics
+	if dom, err = httpDomResponse(urlTopicList); err != nil {
+		return
+	}
+
+	if totalMessages, err = getTotalTopics(dom); err != nil {
+		err = fmt.Errorf("Error getting the total expected topics: %v", err)
+		return
+	}
+
+	if worker > totalMessages/100 {
+		worker = totalMessages / 100
+	}
+
+	topicURLJobs := make(chan string, totalMessages/100+1)
+	results := make(chan urlResults, totalMessages/100+1)
+	defer close(results)
+
+	for i := 0; i < worker; i++ {
+		go getRawMsgURLWorker(org, groupName, topicURLJobs, results)
+	}
+
```

Comment:
Ah this is that pagination part? Maybe a short comment here about the logic also